### PR TITLE
feat(source-declarative-manifest): add components.py support to SDM

### PR DESCRIPTION
**Pull request by bnchrch**
## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by bnchrch** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/43345?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#43345** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/43345?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#39907** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/39907?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#42879** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/42879?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bnchrch and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-integrations/connectors/source-pokeapi-manifest/manifest.yaml [hunk 1]
```diff
@@ -13,22 +13,34 @@ definitions:
...
         type: InlineSchemaLoader
         schema:
           $ref: "#/schemas/pokemon"
+      transformations:
+        - class_name: "source_declarative_manifest.components.ListAddFields"
+          fields:
+            - path: ["bens_duplicated_field"]
+              value: "{{ record['name'] }}"
+              value_type: "string"
+        - class_name: "source_declarative_manifest.components.ListAddFields"
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-pokeapi-manifest/manifest.yaml:39]: The one interesting piece I was wondering about is how can we resolve just `components.%NAME%` into the file, omitting `source_declarative_manifest`. It's probably a small patch to sys paths?

Comment (bnchrch) [reply airbyte-integrations/connectors/source-pokeapi-manifest/manifest.yaml:39]: Oh im sure we can do that easily! Im not sure why we havent tbh!

**Comment by aaronsteers** [comment]
@bnchrch - For my info, can you help me understand the config spec for the version of `source-declarative-manifest` that also picks up `components.py`. Do I understand correctly the behavior here is that we accept a `manifest.yaml` file path and then look for `components.py` as a sibling in the same directory?

I found this [spec](https://github.com/airbytehq/airbyte/blob/c459c8cd119618b35e26fa16647cd95f4330d0d1/airbyte-integrations/connectors/source-declarative-manifest/source_declarative_manifest/spec.json), which only accepts a single input prefixed with a double-underscore. Should we be accepting a directory, or a separate input for path to `components.py`?

**Comment by bnchrch** [comment]
@aaronsteers So we dont have an update yet that will let us pass a filepath for a `components.py` file in a shared volume to a running `sdm` container. 

Nor do we have the ability to use the contents of components.py as an argument to the container

(This is the pattern that exists currently for manifest files and how we power builder project connectors today)

But my (naive) thinking was we would just match this manifest.yaml behaviour for components.py

For the most part the only thing I have strong feelings about at this time is that we enforce end users can only give us one file.

File: airbyte-integrations/connectors/source-pokeapi-manifest/manifest.yaml (hunk 1 above)
Comment (girarda) [reply airbyte-integrations/connectors/source-pokeapi-manifest/manifest.yaml:39]: We can certainly access to module name.

There is some prior art
- https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_file_schema_loader.py#L20-L24
- https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/utils/schema_helpers.py#L132

Happy to chat @ChristoGrab !

**Pull request closed without merging**